structure/set/unionfind.go

```diff
@@ -0,0 +1,63 @@
+package set
+
+// https://en.wikipedia.org/wiki/Disjoint-set_data_structure
+
+type Value interface{}
+
+type node struct {
+	rank  int
+	value Value
+}
+
+type UFS struct {
+	nodes map[Value]*node
+}
+
+func NewUFS() *UFS {
+	return &UFS{
+		nodes: make(map[Value]*node),
+	}
+}
```

Comment:
Make `UFS` unexported since you're setting up the nodes through the `NewUFS` constructor and returning a pointer.